### PR TITLE
[Merged by Bors] - feat(Order/SuccPred/Basic): relax typeclass assumptions on `IsWellOrder.toIsPredArchimedean`

### PR DESCRIPTION
**Pull request by Command-Master**
Make `IsWellOrder.toIsPredArchimedean` and `IsWellOrder.toIsSuccArchimedean` work with non-linear orders.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/SuccPred/Basic.lean [hunk 1]
```diff
@@ -1369,19 +1369,19 @@ instance (priority := 100) IsWellOrder.toIsPredArchimedean [h : IsWellOrder α (
...
-    · cases (min_of_le_pred hb).not_lt hab
+    rcases eq_or_lt_of_le (pred_le b) with hb | hb
+    · cases (min_of_le_pred hb.ge).not_lt hab
     dsimp at ih
     obtain ⟨k, hk⟩ := ih (pred b) hb (le_pred_of_lt hab)
     refine ⟨k + 1, ?_⟩
     rw [iterate_add_apply, iterate_one, hk]⟩
 
-instance (priority := 100) IsWellOrder.toIsSuccArchimedean [h : IsWellOrder α (· > ·)]
+instance (priority := 100) IsWellFounded.toIsSuccArchimedean [h : IsWellFounded α (· > ·)]
```
Comment (YaelDillies) [inline Mathlib/Order/SuccPred/Basic.lean:1379]:
```suggestion
instance (priority := 100) IsWellFounded.toIsSuccArchimedean [h : WellFoundedGT α]
```

File: Mathlib/Order/SuccPred/Basic.lean [hunk 2]
```diff
@@ -1356,11 +1356,11 @@ lemma StrictAnti.not_bddBelow_range [NoMaxOrder α] [PredOrder β] [IsPredArchim
...
 end bdd_range
 
-section IsWellOrder
+section IsWellFounded
 
-variable [LinearOrder α]
+variable [PartialOrder α]
 
-instance (priority := 100) IsWellOrder.toIsPredArchimedean [h : IsWellOrder α (· < ·)]
+instance (priority := 100) IsWellFounded.toIsPredArchimedean [h : IsWellFounded α (· < ·)]
```
Comment (YaelDillies) [inline Mathlib/Order/SuccPred/Basic.lean:1363]:
```suggestion
instance (priority := 100) IsWellFounded.toIsPredArchimedean [h : WellFoundedLT α]
```

**Review by YaelDillies — COMMENTED** [review]
Well spotted!

**Review by YaelDillies — APPROVED** [review]
maintainer merge

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**